packages/sdk-browser/src/sdk-client.ts

```diff
@@ -207,9 +248,39 @@ export class TurnkeyBrowserClient extends TurnkeySDKClientBase {
         sessionExpiry: Number(readOnlySessionResult!.sessionExpiry),
       },
     };
+
     await setStorageValue(StorageKeys.CurrentUser, currentUser);
+
     return readOnlySessionResult!;
   };
+
+  /**
+   * Creates a read-write session. This method infers the current user's organization ID and target userId. To be used in conjunction with an `iframeStamper`: the resulting session's credential bundle can be injected into an iframeStamper to create a session that enables both read and write requests.
+   *
+   * @param email
+   * @param targetEmbeddedKey
+   * @param expirationSeconds
+   * @returns {Promise<SdkApiTypes.TCreateReadWriteSessionResponse>}
+   */
+  loginWithReadWriteSession = async (
+    targetEmbeddedKey: string,
+    expirationSeconds: string = DEFAULT_SESSION_EXPIRATION,
+    userId?: string
+  ): Promise<SdkApiTypes.TCreateReadWriteSessionResponse> => {
+    const readWriteSessionResult = await this.createReadWriteSession({
+      targetPublicKey: targetEmbeddedKey,
+      expirationSeconds,
+      userId: userId!,
+    });
+
+    // store auth bundle in local storage
+    await setStorageValue(StorageKeys.ReadWriteSession, {
+      authBundle: readWriteSessionResult!.credentialBundle,
+      sessionExpiry: Date.now() + Number(expirationSeconds) * 1000,
+    });
+
+    return readWriteSessionResult;
+  };
```

Comment:
@andrewkmin I assume I could use this method like this? 
```
authIframeClient.loginWithReadWriteSession(authIframeClient?.iframePublicKey)
```
if the authIframeClient already has this public key I wonder if this method could be on the `TurnkeyIframeClient` so it doesn't have to be passed in?

Reply:
> authIframeClient.loginWithReadWriteSession(authIframeClient?.iframePublicKey)

Yeah that's exactly right!

> if the authIframeClient already has this public key I wonder if this method could be on the TurnkeyIframeClient so it doesn't have to be passed in?

Currently, any of the clients can initiate a `login with read write session` activity (e.g. `passkeyClient.loginWithReadWriteSession` is valid as well). You're right in that the TurnkeyIframeClient could override the base implementation (that lives on `TurnkeyBrowserClient`) and use something like `this.iframePublicKey()`

Reply:
tightt that would be so slicckkk!